packages/sdk-drop-in/src/lib/useIframeAppSlots.tsx

```diff
@@ -0,0 +1,86 @@
+import { ComponentProps, useCallback, useEffect, useState } from 'react';
+
+import { MoniteIframeAppCommunicator } from '@/lib/MoniteIframeAppCommunicator';
+import { MoniteProvider } from '@monite/sdk-react';
+
+type AccessToken = {
+  access_token: string;
+  token_type: string;
+  expires_in: number;
+};
+
+export const useMoniteIframeAppSlots = () => {
+  const [slots, setSlots] = useState<
+    Partial<Pick<ComponentProps<typeof MoniteProvider>, 'locale' | 'theme'>>
+  >({});
+
+  const [iframeCommunicator] = useState(
+    () => new MoniteIframeAppCommunicator(window.parent)
+  );
+
+  const fetchToken = useCallback(() => {
+    return new Promise<AccessToken>((resolve) => {
+      iframeCommunicator.on('fetch-token', (data) => {
+        if (!validateToken(data))
+          throw new Error('Invalid token received from the iframe');
+
+        resolve(data);
+      });
+      iframeCommunicator.pingSlot('fetch-token');
+    }).finally(() => {
+      iframeCommunicator.off('fetch-token');
+    });
+  }, [iframeCommunicator]);
+
+  useEffect(
+    function subscribe() {
+      iframeCommunicator.on('locale', (locale) => {
+        // @ts-expect-error - payload is not a valid theme object
+        setSlots((prevSlots) => ({ ...prevSlots, locale }));
+      });
+
+      iframeCommunicator.on('theme', (theme) => {
+        // @ts-expect-error - payload is not a valid theme object
+        setSlots((prevSlots) => ({ ...prevSlots, theme }));
+      });
+
+      iframeCommunicator.connect();
+
+      return () => {
+        iframeCommunicator.disconnect();
+      };
+    },
+    [iframeCommunicator, fetchToken]
+  );
+
+  useEffect(
```

Comment:
I think it is possible to skip using `clicker` and use `fetchToken` directly.

```ts
useEffect(
  function subscribe() {
    document.addEventListener('click', fetchToken);

    return () => {
      document.removeEventListener('click', fetchToken);
    };
  },
  [fetchToken]
);
```

Reply:
Thank you, removed this garbage. This is why I asked for the PR.